basestar-database-server/src/test/java/io/basestar/database/TestDatabaseServer.java

```diff
@@ -131,12 +132,16 @@ void setUp() throws Exception {
         );
         this.emitter = Mockito.mock(Emitter.class);
         when(emitter.emit(any(Event.class))).then(inv -> {
-            log.info("Emitting {}", inv.getArgumentAt(0, Event.class));
+            log.info("Emitting {}", inv.getArgument(0, Event.class));
             return CompletableFuture.completedFuture(null);
         });
         when(emitter.emit(any(Collection.class))).then(inv -> {
             final Emitter emitter = (Emitter) inv.getMock();
-            inv.getArgumentAt(0, Collection.class).forEach(event -> emitter.emit((Event) event));
+            final Collection<Event> events = inv.getArgument(0, Collection.class);
+            log.info("Got collection of events with size: {}", events.size());
+            events.forEach(emitter::emit);
+            log.info("..Done emitting collection of events with size: {}", events.size());
```

Comment:
(extra logging to be dropped once behaviour around line 286 is confirmed correct)